backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.images;
+
+import static org.mockito.Mockito.*;
+import com.google.protobuf.ByteString;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.IllegalArgumentException;
+import java.io.IOException;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import java.io.ByteArrayOutputStream;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.Arrays;
+import javax.imageio.ImageIO;
+import java.lang.IllegalArgumentException;
+import com.google.sps.images.data.AnnotatedImage;
+import org.junit.Before;
+
+/**
+ * Tests for AnnotatedImage.
+ */
+public final class AnnotatedImageTest {
+
+  // Testing strategy for each operation of AnnotatedImage:
+  /*
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  Partition on rawImageData: null, empty, non-empty image data.
+   *  Partition on labelAnnotations: empty, non-empty image data.
+   * 
+   * public String getLabelsAsJson();
+   *  Unnecesary to test as it is a wrapper for a Gson call.
+   *
+   * public List<String> getLabelDescriptions();
+   *  Partition on this: labelAnnotations is empty, or non-empty list of EntityAnnotations.
+   */
+
+  private byte[] nullRawImageData;
+  private byte[] emptyRawImageData;
+  private byte[] rawImageData;
+  private List<EntityAnnotation> emptyLabelAnnotations;
+  private List<EntityAnnotation> labelAnnotations;
```

Comment:
Minimizing state in tests is preferred (so reader doesn't need to wonder if this is getting modified by some side effect during a test). If any of these can be converted to constants, or helper methods, those are preferred, e.g.:

```
/** Label annotations with arbitrary elements. */
List<EntityAnnotation> makeNonEmptyLabelAnnotations() {
  List<EntityAnnotation> labelAnnotations = new ArrayList<>();

    EntityAnnotation entityAnnotationMock = mock(EntityAnnotation.class);
    // mock the behavior of stock service to return the value of various stocks
    when(entityAnnotationMock.getDescription()).thenReturn("arbitrary description");  // this description should not be referred to from tests
    labelAnnotations.add(entityAnnotationMock);
    labelAnnotations.add(entityAnnotationMock);
    return labelAnnotations;
}

void testFoo() {
  List<EntityAnnotation> labelAnnotations = makeNonEmptyLabelAnnotations();
  // Do something with labelAnnotations
}
```

Reply:
These actually are all constants, I should have declared them final. I will update this.